config.py

```diff
@@ -1,72 +1,56 @@
-# coding: utf8
-"""
-config.py - Misc constants
-
-Copyright (c) 2018 The Fuel Rats Mischief,
-All rights reserved.
-
-Licensed under the BSD 3-Clause License.
-
-See LICENSE.md
-
-This module is built on top of the Pydle system.
-
-"""
+import json
 import logging
-
-
-class IRC:
-    """
-    IRC Configuration
-    """
-    ####
-    # Mecha's presence in IRC
-    presence = "Mechasqueak3-unknown[BOT]"
-    ####
-    # Server to connect to
-    server = "dev.localecho.net"
-    ####
-    # Port to connect on
-    port = "6667"
-    ####
-    # use TLS
-    tls = False
-    ####
-    # what channels to connect to
-    channels = ["#unkn0wndev"]
-
-    class Authentication:
-        """
-        Bots Authentication configuration
-
-        Currently this is just a skeleton and has no effect.
-        """
-        # TODO: Implement SASL authentication
-        username = ""
-        password = ""
-        SASL = True
-
-
-class Logging:
-    """
-    Log configurations
+import os
+from typing import Union
+
+CONFIGURATION: Union[None, dict] = None
+
+
+def setup_logging(root_logger: str, logfile: str):
+    # create a log formatter
+    log_formatter = logging.Formatter("{levelname} [{name}::{funcName}]:{message}",
+                                      style='{')
+    # get Mecha's root logger
+    log = logging.getLogger(root_logger)
+    # Create a file handler for the logger
+    log_file_handler = logging.FileHandler(logfile, 'w')
+    log_file_handler.setFormatter(log_formatter)
+    # create a stream handler ( prints to STDOUT/STDERR )
+    log_stream_handler = logging.StreamHandler()
+    log_stream_handler.setFormatter(log_formatter)
+    # adds the two handlers to the logger so they can do their thing.
+    log.addHandler(log_file_handler)
+    log.addHandler(log_stream_handler)
+    # set the minimum severity the logger will report.
+    # uncomment for production:
+    # log.setLevel(logging.INFO)
+    # uncomment for develop:
+    log.setLevel(logging.DEBUG)
+
+    logging.info("[Mecha] configuration file loading...")
+    logging.basicConfig(level=logging.DEBUG)  # write all the things
+
+    """provides facilities for managing a configuration from disk"""
```

Comment:
Stray docstring?

Reply:
Uhh... that was supposed to be at the top of file. So yes, stray. Will correct.